mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts

```diff
@@ -28,39 +71,49 @@ export class WayPreviewDAL {
   public static async getWaysPreviewByUuids(wayUuids: string[]): Promise<WayPreview[]> {
     const waysDTO = wayUuids.length !== 0 ? await WayService.getWaysDTOByUuids(wayUuids) : [];
 
-    const waysPreview = Promise.all(waysDTO.map(async (wayDTO) => {
-      const ownerPromise = UserPreviewDAL.getUserPreview(wayDTO.ownerUuid);
+    const allNeededUsersUuids = new Set(waysDTO.flatMap(wayDTO => [
+      wayDTO.ownerUuid,
+      ...wayDTO.mentorUuids,
+      ...wayDTO.mentorRequestUuids,
+    ]));
 
-      const mentorsPromise = Promise.all(wayDTO.mentorUuids.map(UserPreviewDAL.getUserPreview));
+    const allNeededGoalsUuids = new Set(waysDTO.flatMap(wayDTO => [wayDTO.goalUuid]));
 
-      const mentorRequestsPromise = Promise.all(wayDTO.mentorRequestUuids.map(UserPreviewDAL.getUserPreview));
+    const allNeededUsersPreview = await UserPreviewDAL.getUsersPreviewByUuids(Array.from(allNeededUsersUuids));
 
-      const goalPromise = GoalPreviewDAL.getGoal(wayDTO.goalUuid);
+    const allNeededGoalsPreview = await GoalPreviewDAL.getGoalsPreviewByUuids(Array.from(allNeededGoalsUuids));
 
-      const [
-        owner,
-        mentors,
-        mentorRequests,
-        goal,
-      ] = await Promise.all([
-        ownerPromise,
-        mentorsPromise,
-        mentorRequestsPromise,
-        goalPromise,
-      ]);
+    const waysPreview = await Promise.all(waysDTO.map(async (wayDTO) => {
+      const owner = allNeededUsersPreview.find(user => user.uuid === wayDTO.ownerUuid);
+      const mentors = wayDTO.mentorUuids.map(uuid => allNeededUsersPreview.find(user => user.uuid === uuid));
+      const mentorRequests = wayDTO.mentorRequestUuids.map(uuid => allNeededUsersPreview.find(user => user.uuid === uuid));
```

Comment:
lets try to get rid of find methods to avoid recheck data for typescript, if possible

Reply:
Changed to filter